lib/src/platform/src/background_android_isolate_platform.dart

```diff
@@ -20,7 +20,8 @@ class BackgroundIsolateAndroidPlatform {
     _methodChannel.setMethodCallHandler((call) async {
       switch (call.method) {
         case PlatformMethod.pushOnBackgroundMessage:
-          return _onPushBackgroundMessage(call.arguments as RemoteMessage);
+          await _onPushBackgroundMessage(call.arguments as RemoteMessage);
```

Comment:
That one I'm fairly sure it needs to be await.  That's the code for the isolate.  When that code ends it ends the isolate.  I believe if it returns it will just return Future<void>, then closes the isolate without fully executing the _onPushBackgroundMessage().  

My first test for this change was to add a delay here, which worked, but wasn't an ideal solution.

Reply:
@asoap yes, but this `async` function and it has some syntax sugar, e.g.  returned `Future` should be unwrapped (basically awaited) automatically, otherwise it would return `Future<Future<void>>`.

Reply:
Can we do it like 
```
      dynamic onBackgroundMessageResult = _onPushBackgroundMessage(call.arguments as RemoteMessage);
      // ^^^ _onBackgroundMessage() can return void or Future<void>
      if (onBackgroundMessageResult is Future) {        
        await onBackgroundMessageResult; // if it returns a Future we await on it.
      }
```
If you are not 100% sure if its a future

Reply:
@ttypic AFAIK, if we have nested promises, generally all of them resolve in one go with `await`. Not sure how it works in flutter though

Reply:
@sacOO7 Futures are flatten here for sure, this code works right now :) and `_onPushBackgroundMessage ` returns `Future<void>` right now, there were no changes, I think you are confused with  `onPushBackgroundMessage` from `lib/src/platform/src/push_notification_events_internal.dart`

Reply:
@ttypic  The _onPushBackgroundMessage() call inside of background_android_isolate_platform.dart calls handleBackgroundMessage() in push_notification_events_internal.dart   That's why I changed the handleBackgroundMessage() to be a Future<void> as well. This is setup in a way where these two functions are doing the same thing.  They both check to see if there is a function to call, and then call it.   The _onPushBackgroundMessage() should always get a Future<void> from handleBackgroundMessage();